Enhance readme, document link checker, link checker script (#549)
substrate-developer-hub/substrate-developer-hub.github.io: readme.md

source code for some top-level pages in `/website/pages/en`. Follow our [contribution guidelines](CONTRIBUTING.md).

### Adding a new document

To add a new markdown document,

* Create your markdown document in a suitable directory inside `/docs`.
* If you have images in your document, put them in the `/docs/assets/` directory.
* Documentation should follow our [contribution guidelines](CONTRIBUTING.md).
* If you want your document to appear in the sidebar, add its reference in the `/website/sidebar.json` file under the
  corresponding section.

### Rename an existing document

To rename an existing document,

* Change the name or path of the document.
* After the change has been merged, go to the [Crowdin project](https://crowdin.com/project/substrate-developer-hub),
  make sure the translation is already migrated to the new file automatically for all the target languages.
* Then go to [Crowdin project settings](https://crowdin.com/project/substrate-developer-hub/settings#files), remove the
  old source file in `Files` tab.
* If you don't have access to the Crowdin project, please send email to <substrate-developer-hub@parity.io> with the

>   file information you want to remove.

Reviewer: Is this still relevant @kaichaosun

you did not introduce, please
[create an Issue](https://github.com/substrate-developer-hub/substrate-developer-hub.github.io/issues/new).

Once you are done with your changes, feel free to submit a PR.

### Staging Deployment

There are two ways to deploy to *staging* environment, which you can access at
[staging.substrate.dev](https://staging.substrate.dev), which is hosted on Heroku. Please check with the team members
for the HTTP basic authentication username and password.

1. Commit to your local repository. Run [`scripts/deploy-staging`](./scripts/deploy-staging). This builds the project
into the docusaurus static site, and force push the static site to `staging` branch. The result is then picked up by
Heroku. Note that multilingual translations are NOT pulled in.

2. Commit to your local repository, and then push to `staging-source` branch. This triggers the CI to build the website
AND also pull in multilingual translations from our Crowdin project. The final built static site is then pushed to
`staging` branch and deployed to Heroku.

### Production Deployment

> Our production site is at [substrate.dev](https://substrate.dev). To deploy to production, push your update in `source`

Reviewer: ```suggestion
Our production site is at [substrate.dev](https://substrate.dev). To deploy to production, merge your update into the `source`
```
